sympy/printing/python.py

```diff
@@ -1,92 +1,29 @@
-            f = getattr(StrPrinter, f_name)
-            setattr(PythonPrinter, f_name, f)
-
-    def _print_Function(self, expr):
-        func = expr.func.__name__
-        if not hasattr(sympy, func) and not func in self.functions:
-            self.functions.append(func)
-        return StrPrinter._print_Function(self, expr)
-
-    # procedure (!) for defining symbols which have be defined in print_python()
-    def _print_Symbol(self, expr):
-        symbol = self._str(expr)
-        if symbol not in self.symbols:
-            self.symbols.append(symbol)
-        return StrPrinter._print_Symbol(self, expr)
-
-    def _print_module(self, expr):
-        raise ValueError('Modules in the expression are unacceptable')
-
-
+from sympy.core.decorators import deprecated
+from .sympycode import SymPyPrinter, sympy_code, print_sympy_code
+
+@deprecated(
+    last_supported_version='1.2',
+    useinstead="SymPyPrinter",
+    issue=12807,
+    deprecated_since_version='1.1')
+class PythonPrinter(SymPyPrinter):
+    pass
+
+
+@deprecated(
+    last_supported_version='1.2',
+    useinstead="SymPyPrinter",
+    issue=12807,
+    deprecated_since_version='1.1')
 def python(expr, **settings):
-    """Return Python interpretation of passed expression
-    (can be passed to the exec() function without any modifications)"""
-
-    printer = PythonPrinter(settings)
-    exprp = printer.doprint(expr)
-
-    result = ''
-    # Returning found symbols and functions
-    renamings = {}
-    for symbolname in printer.symbols:
-        newsymbolname = symbolname
-        # Escape symbol names that are reserved python keywords
-        if kw.iskeyword(newsymbolname):
-            while True:
-                newsymbolname += "_"
-                if (newsymbolname not in printer.symbols and
-                        newsymbolname not in printer.functions):
-                    renamings[sympy.Symbol(
-                        symbolname)] = sympy.Symbol(newsymbolname)
-                    break
-        result += newsymbolname + ' = Symbol(\'' + symbolname + '\')\n'
-
-    for functionname in printer.functions:
-        newfunctionname = functionname
-        # Escape function names that are reserved python keywords
-        if kw.iskeyword(newfunctionname):
-            while True:
-                newfunctionname += "_"
-                if (newfunctionname not in printer.symbols and
-                        newfunctionname not in printer.functions):
-                    renamings[sympy.Function(
-                        functionname)] = sympy.Function(newfunctionname)
-                    break
-        result += newfunctionname + ' = Function(\'' + functionname + '\')\n'
-
-    if not len(renamings) == 0:
-        exprp = expr.subs(renamings)
-    result += 'e = ' + printer._str(exprp)
-    return result
+    return sympy_code(expr, **settings)
 
 
+@deprecated(
+    last_supported_version='1.2',
+    useinstead="SymPyPrinter",
```

Comment:
Replacement should be `sympy_code`, not `sympy_printer`. Also, should use the full path name since it isn't imported by default (`sympy.printing.sympycode.sympy_code`).